routing_common/transit_types.hpp

```diff
@@ -104,6 +126,76 @@ class Edge
   std::vector<ShapeId> m_shapeIds;
 };
 
-// @TODO(bykoianko) Data structures and methods for other transit data should be implemented in here.
+class Transfer
+{
+public:
+  Transfer() = default;
+  Transfer(StopId id, m2::PointD const & point, std::vector<StopId> const & stopIds);
+  bool IsEqualForTesting(Transfer const & transfer) const;
+
+  DECLARE_VISITOR_AND_DEBUG_PRINT(Transfer, visitor(m_id, "id"), visitor(m_point, "point"),
+                                  visitor(m_stopIds, "stop_ids"))
+
+private:
+  StopId m_id = kInvalidStopId;
+  m2::PointD m_point;
+  std::vector<StopId> m_stopIds;
+
+  // @TODO(bykoianko) It's necessary to add field m_titleAnchors here and implement serialization
+  // and deserialization.
+};
+
+class Line
+{
+public:
+  Line() = default;
+  Line(LineId id, std::string const & number, std::string const & title, std::string const & type,
+       NetworkId networkId, std::vector<StopId> const & stopIds);
+  bool IsEqualForTesting(Line const & line) const;
+
+  DECLARE_VISITOR_AND_DEBUG_PRINT(Line, visitor(m_id, "id"), visitor(m_number, "number"),
+                                  visitor(m_title, "title"), visitor(m_type, "type"),
+                                  visitor(m_networkId, "network_id"),
+                                  visitor(m_stopIds, "stop_ids"))
+
+private:
+  LineId m_id = kInvalidLineId;
+  std::string m_number;
+  std::string m_title;
+  std::string m_type;
+  NetworkId m_networkId = kInvalidNetworkId;
+  std::vector<StopId> m_stopIds;
+};
+
+class Shape
+{
+public:
+  Shape() = default;
+  Shape(ShapeId id, StopId stop1_id, StopId stop2_id, std::vector<m2::PointD> const & polyline);
```

Comment:
Could you please try Encode/Decode polyline for Shape?

Reply:
Yes. I'll do that. Pls have a look at my comment above.